src/transformers/trainer.py

```diff
@@ -2104,7 +2113,7 @@ def _load_from_checkpoint(self, resume_from_checkpoint, model=None):
                     "yield to errors or unwanted behaviors."
                 )
 
-        if os.path.isfile(weights_file) or os.path.isfile(safe_weights_file):
+        if os.path.isfile(weights_file) or os.path.isfile(safe_weights_file) or is_fsdp_ckpt:
```

Comment:
just wondering here, we could have `is_fsdp_enabled` set to `False` but want to load from a `fsdp` checkpoint. Little edge case that we can prevent!

Because this function does not seem to account for that!